java/arcs/core/entity/StorageAdapter.kt

```diff
@@ -27,6 +30,29 @@ sealed class StorageAdapter<T : Storable, R : Referencable> {
 
     /** Checks if the [Storable] is expired (its expiration time is in the past). */
     abstract fun isExpired(value: T): Boolean
+
+    @Throws(IllegalStateException::class)
```

Comment:
This isn't really necessary unless we want java callers to have to catch the exception.